src/runtime/GlobalObjectBuiltinArray.cpp

```diff
@@ -1498,6 +1537,40 @@ static Value builtinArrayPush(ExecutionState& state, Value thisValue, size_t arg
     return Value(n);
 }
 
+// https://www.ecma-international.org/ecma-262/10.0/#sec-array.prototype.flat
+// Array.prototype.flat( [ depth ] )
+static Value builtinArrayFlat(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    RESOLVE_THIS_BINDING_TO_OBJECT(O, Array, flat);
+    int64_t sourceLen = O->lengthES6(state);
+    double depthNum = 1;
+    if (argc > 0 && !argv[0].isUndefinedOrNull()) {
```

Comment:
Oops, its a typo. We should check undefined type so `isUndefined()` would be suitable. Others are fine:)